### PR TITLE
Added multiple binding support in IIS Manage

### PR DESCRIPTION
**Pull request by asranja**

### CODE REVIEW

File: Tasks/IISWebAppManagementOnMachineGroup/Tests/L0.ts [hunk 1]
```diff
@@ -27,55 +27,6 @@ describe('IISWebAppManagementOnMachineGroup Suite', function () {
     });
 
     if (ps) {
-
```
Comment (asranja) [inline Tasks/IISWebAppManagementOnMachineGroup/Tests/L0.ts]: Moved the tests for the common script to the vsts-rm-extensions repo

File: Tasks/IISWebAppManagementOnMachineGroup/Utility.ps1 [hunk 2]
```diff
@@ -230,41 +246,84 @@ function Repair-Inputs([ref]$siteName, [ref]$physicalPath, [ref]$poolName, [ref]
...
         {
-            Write-Warning (Get-VstsLocString -Key "SSLCertWarningInvalidCharacters")
+            Write-Warning (Get-VstsLocString -Key "SSLCertWarningInvalidCharactersInBinding" -ArgumentList $protocol, $ipAddress, $port)
         }
 
-        $sslCertThumbprint.Value = [Regex]::Replace($sslCertThumbprint.Value, "[^a-fA-F0-9]+" , "")
-
+        $sslCertThumbPrint.Value = [Regex]::Replace($sslCertThumbPrint.Value, "[^a-fA-F0-9]+" , "")
+        
         # Mark the SSL thumbprint value to be a secret value
```
Comment (kmkumaran) [inline Tasks/IISWebAppManagementOnMachineGroup/Utility.ps1]: This comment is not relevant for below newly introduced statements. You can move the comment closer to the below code where we set secret variable.

File: Tasks/IISWebAppManagementOnMachineGroup/Utility.ps1 [hunk 3]
```diff
@@ -230,41 +246,84 @@ function Repair-Inputs([ref]$siteName, [ref]$physicalPath, [ref]$poolName, [ref]
...
+    
+    $bindingsObj = $bindings | ConvertFrom-Json 
+
+    foreach ($binding in $bindingsObj.bindings) {
+
+        if($binding.protocol -eq "https") {
+            if(-not [string]::IsNullOrWhiteSpace($binding.sslThumbprint)) {
+                # Trim all non-hexadecimal characters from the ssl cetificate thumbprint
+                $sslCertThumbPrint = $binding.sslThumbPrint
+                $sslCertThumbPrint = Repair-Inputs -sslCertThumbPrint ([ref]$sslCertThumbPrint) -ipAddress $binding.ipAddress -protocol $binding.protocol -port $binding.port
```
Comment (kmkumaran) [inline Tasks/IISWebAppManagementOnMachineGroup/Utility.ps1]: except cert thumbprint, other inputs are not really inputs for repairing. The Name of the method and signature is not matching.

File: Tasks/IISWebAppManagementOnMachineGroup/task.json [hunk 4]
```diff
@@ -221,52 +224,56 @@
...
         {
+            "name": "Bindings",
+            "type": "multiLine",
+            "label": "Add bindings",
+            "required": true,
+            "groupName": "IISBindings",
+            "properties": {
+                "resizable": "true",
+                "editorExtension": "ms.vss-services-azure.iis-multiple-binding",
+                "displayFormat": "{{#bindings}} {{protocol}}:{{ipAddress}}:{{port}}:{{hostname}} \n{{/bindings}}"
```
Comment (kmkumaran) [inline Tasks/IISWebAppManagementOnMachineGroup/task.json]: Is it possible to have same semantics as in inetmgr.

**Pull request merged**